src/components/ui/Text/Text.tsx

```
@@ -8,16 +8,23 @@ import { customClassSwitcher } from '~/core';
     customRootClass?: string;
     className?: string;
+    as?: 'div' | 'span' | 'p' | 'label';
 } & React.ComponentProps<'p'>;
 
-const Text = ({ children, customRootClass = '', className = '', ...props }: TextProps) => {
+const Text = ({ children, customRootClass = '', className = '', as='p' , ...props }: TextProps) => {
     const rootClass = customClassSwitcher(customRootClass, COMPONENT_NAME);
-
-    return <p className={`${rootClass} ${className}`} {...props}>{children}</p>;
+      let Tag = 'p'
+      tagMap.map((tag) => tag === as && (Tag = as)) ;
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)